oauthproxy.go

```diff
@@ -265,7 +265,7 @@ func (p *OAuthProxy) setupServer(opts *options.Options) error {
 }
 
 func (p *OAuthProxy) buildServeMux(proxyPrefix string) {
-	r := mux.NewRouter()
+	r := mux.NewRouter().UseEncodedPath()
```

Comment:
Worth adding a comment here to point out that we use the encoded path here to make sure the decision can be made at the second mux/proxy level whether to send the encoded path to the upstream servers? (So we know why when we look at the code in 6 months time)

Reply:
agreed. i will add a comment